src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/VisualStudioNuGetProject.cs

```
@@ -0,0 +1,125 @@
+        }
+
+        public async override Task<(bool, object)> TryGetMetadataAsync(string key, CancellationToken token)
+        {
+            var remoteBroker = await BrokeredServicesUtilities.GetRemoteServiceBrokerAsync();
+            using (var nugetProjectManagerService = await remoteBroker.GetProxyAsync<INuGetProjectManagerService>(NuGetServices.ProjectManagerService, cancellationToken: token))
+            {
+                Assumes.NotNull(nugetProjectManagerService);
+
+                return await nugetProjectManagerService.TryGetMetadataAsync(_projectGuidString, key, token);
+            }
+        }
+
+        public async override Task<object> GetMetadataAsync(string key, CancellationToken token)
+        {
+            var remoteBroker = await BrokeredServicesUtilities.GetRemoteServiceBrokerAsync();
+            using (var nugetProjectManagerService = await remoteBroker.GetProxyAsync<INuGetProjectManagerService>(NuGetServices.ProjectManagerService, cancellationToken: token))
+            {
+                Assumes.NotNull(nugetProjectManagerService);
+
+                var metadataResult = await nugetProjectManagerService.GetMetadataAsync(_projectGuidString, key, token);
+                return metadataResult;
+            }
+        }
+
+        public static string GetProjectGuidStringFromVslsQueryString(string queryString)
+        {
+            var result = Guid.Empty;
+
+            try
+            {
+                Uri pathUri = new Uri(queryString);
+
+                if (string.Equals(pathUri.Scheme, LiveShareUriScheme, StringComparison.OrdinalIgnoreCase))
+                {
+                    var queryStrings = ParseQueryString(pathUri);
+                    if (!queryStrings.TryGetValue(ProjectGuidQueryString, out var projectGuid) || !Guid.TryParse(projectGuid, out result))
+                    {
+                        result = Guid.Empty;
+                    }
+                }
+            }
+            catch
```

We shouldn't catch all exceptions.  You should catch specific exceptions.

You can avoid exceptions from parsing the query string by using `Uri.TryCreate(...)` instead, similar to how you used `Guid.TryParse(...)` to avoid parsing exceptions.

You may find that you don't need a `try...catch` block or that the `try` block can be narrowed in scope.